src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1381,6 +1351,57 @@ IBinaryOperation createBoundUserDefinedConditionalLogicalOperator(BoundUserDefin
             }
         }
 
+        private IBinaryOperation CreateBoundBinaryOperatorOperation(BoundBinaryOperator boundBinaryOperator, IOperation left, IOperation right)
+        {
+            BinaryOperatorKind operatorKind = Helper.DeriveBinaryOperatorKind(boundBinaryOperator.OperatorKind);
+            IMethodSymbol? operatorMethod = boundBinaryOperator.Method.GetPublicSymbol();
+            IMethodSymbol? unaryOperatorMethod = null;
+
+            // For dynamic logical operator MethodOpt is actually the unary true/false operator
+            if (boundBinaryOperator.Type.IsDynamic() &&
+                (operatorKind == BinaryOperatorKind.ConditionalAnd || operatorKind == BinaryOperatorKind.ConditionalOr) &&
+                operatorMethod?.Parameters.Length == 1)
+            {
+                unaryOperatorMethod = operatorMethod;
+                operatorMethod = null;
+            }
+
+            SyntaxNode syntax = boundBinaryOperator.Syntax;
+            ITypeSymbol? type = boundBinaryOperator.GetPublicTypeSymbol();
+            ConstantValue? constantValue = boundBinaryOperator.ConstantValue;
+            bool isLifted = boundBinaryOperator.OperatorKind.IsLifted();
+            bool isChecked = boundBinaryOperator.OperatorKind.IsChecked();
+            bool isCompareText = false;
+            bool isImplicit = boundBinaryOperator.WasCompilerGenerated;
+            return new BinaryOperation(operatorKind, left, right, isLifted, isChecked, isCompareText, operatorMethod, unaryOperatorMethod,
+                                       _semanticModel, syntax, type, constantValue, isImplicit);
+        }
+
+        private IOperation CreateBoundInterpolatedStringBinaryOperator(BoundBinaryOperator boundBinaryOperator)
+        {
+            Debug.Assert(boundBinaryOperator.InterpolatedStringHandlerData is not null);
+            Func<(CSharpOperationFactory, InterpolatedStringHandlerData), BoundInterpolatedString, int, IOperation> createInterpolatedString
+                = createInterpolatedStringOperand;
+
+            Func<(CSharpOperationFactory, InterpolatedStringHandlerData), BoundBinaryOperator, IOperation, IOperation, IOperation> createBinaryOperator
+                = createBoundBinaryOperatorOperation;
+
+            return boundBinaryOperator.RewriteInterpolatedStringAddition((this, boundBinaryOperator.InterpolatedStringHandlerData.GetValueOrDefault()), createInterpolatedString, createBinaryOperator);
+
+            static IInterpolatedStringOperation createInterpolatedStringOperand(
+                (CSharpOperationFactory @this, InterpolatedStringHandlerData Data) arg,
```

Comment:
>@this [](http://example.com/codeflow?start=40&length=5)

nit: could be `This` given naming convention on tuple elements

Reply:
It could be, but I prefer `@this`.